lib/internals.js

```diff
@@ -837,10 +839,15 @@ module.exports = (self) => {
      */
     async getAposViteConfig(baseConfig) {
       const vite = await import('vite');
-      const aposConfig = await viteAposConfig({
+      const config = await viteAposConfig({
         sourceRoot: self.buildRootSource,
         input: self.getBuildInputs('apos')
       });
+      const postcssConfig = await vitePostcssConfig({
```

Comment:
A better implementation of this would be:
- Move the entire postcss config processing in its own `async getPostcssConfig(options)` method.
- Inside of this method, detect the `apos.assset.options` and just send simple instructions to the config. Is the plugin enabled? Pass the configuration options that the config may pass directly to the plugin. Or even better, use the build options (see below) - they are cached in `self.buildOptions`.
- What I think is really bad here, is that Vite module now knows Apostrophe core internals. Something I wanted to avoid. Vite module only cares about building. It shouldn't know anything about "mobile preview" or any other feature and business logic. We should decouple. So I really recommend to patch the core, find `getBuildOptions()` core asset method, and add additional build options, something like:
```js
{
  postcssTransformViewport: {
  enable: self.options.breakpointPreviewMode?.enable,
  debug: self.options.breakpointPreviewMode?.debug === true,
  attr: 'data-breakpoint-preview-mode',
  hook: self.options.breakpointPreviewMode?.transform
}
```
Pass those to `await getPostcssConfig({ ...self.buildOptions.postcssTransformViewport  })`
- Remove all the business logic from `vite-postcss-config.js`. The config should only check if the plugin is needed, if yes load it and pass blindly the arguments. It should not perform any merges as well, a plain object containing only the known postcss plugin.
- Back to `getPostcssConfig`, it loads the base postcss config using the factory function as it does now (but with the simple options described above). It also should take care of detecting the project level postcss config file (if any). And this is a no small task! More about it below. In the end it should use the Vite merge helper to merge our base config and loaded postcss config:
```js
const finalPostcssConfig = await vite.mergeConfig(baseConfig, {
  css: {
    postcss: userConfig
  }
});
```
- Loading the project level postcss config might be its own method as well. Reference to vite docs: https://vite.dev/config/shared-options.html#css-postcss. It shows that Vite itself attempts loading the config using a plugin - https://github.com/postcss/postcss-load-config?tab=readme-ov-file. And it's a pretty complex logic! I think our best move here is to use internally this plugin (it exports a single async function) and pass the proper configuration to it. It's pretty simple code, so implement this shouldn't be a problem. Note - use `self.apos.rootDir` as a base config location, because `process.cwd()` will break multisite.
- In the end, make sure that loading user (project level) postcss config in `getPostcssConfig` happens conditionally based on an option. Switch this option off in `getAposViteConfig`, and on in `getPublicViteConfig`.
- In the end, I would also recommend make the postcss plugin ESM module. I'm not sure what are the reasons for you to go back to CJS, but using a dynamic `import()` inside of the config file is OK and already done for our own plugin. The config factories are all async by design.